### PR TITLE
PEP: pandas/core round 4 (indexing, internals, missing)

### PR DESCRIPTION
**Pull request by rockg**

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -206,7 +216,8 @@ def _setitem_with_indexer(self, indexer, value):
         self._has_valid_setitem_indexer(indexer)
 
         # also has the side effect of consolidating in-place
-        from pandas import Panel, DataFrame, Series
+        # TODO: Panel, DataFrame are not imported, remove?
```
Comment (jreback) [inline pandas/core/indexing.py]: why is this added? these are not imported because it would be a circluar import otherwise (the ``NDFrame`` objects need to import the indexers to make them properties).

**Comment by jreback** [comment]
minor comment. lgtm

File: pandas/core/indexing.py (hunk 1 above)
Comment (rockg) [inline pandas/core/indexing.py]: Should read "are not used".  They aren't used anywhere.

**Comment by wesm** [comment]
thanks @rockg ! I can attest this is not the most pleasant sort of work
